frontend/src/components/Issue/PipelineGhostFlow.vue

```diff
@@ -0,0 +1,365 @@
+                  <span class="text-xs scale-90">{{ percent }}%</span>
+                </template>
+              </BBProgressPie>
+            </div>
+          </div>
+
+          <div
+            v-if="j < taskList.length - 1"
+            class="hidden md:flex items-center justify-center w-4 h-2 overflow-visible relative"
+          >
+            <!-- show an arrow indicator between tasks -->
+            <heroicons-outline:chevron-right class="w-4 h-4" />
+          </div>
+        </template>
+      </div>
+
+      <div class="absolute right-0 top-0 bottom-0 w-10 hidden md:block"></div>
+    </NScrollbar>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { computed, PropType, watchEffect } from "vue";
+import {
+  Pipeline,
+  Stage,
+  StageCreate,
+  PipelineCreate,
+  Task,
+  TaskCreate,
+  Project,
+  Database,
+} from "@/types";
+import { activeTask, activeTaskInStage, taskSlug } from "@/utils";
+import { isEmpty } from "lodash-es";
+import TaskStatusIcon from "./TaskStatusIcon.vue";
+import { useDatabaseStore } from "@/store";
+import { NScrollbar } from "naive-ui";
+import { BBProgressPie } from "@/bbkit";
+import { useI18n } from "vue-i18n";
+
+const props = defineProps({
+  create: {
+    required: true,
+    type: Boolean,
+  },
+  project: {
+    required: true,
+    type: Object as PropType<Project>,
+  },
+  pipeline: {
+    required: true,
+    type: Object as PropType<Pipeline | PipelineCreate>,
+  },
+  selectedStage: {
+    required: true,
+    type: Object as PropType<Stage | StageCreate>,
+  },
+  selectedTask: {
+    type: Object as PropType<Task | TaskCreate | undefined>,
+    default: undefined,
+  },
+});
+
+const emit = defineEmits<{
+  (e: "select-task", stageIdOrIndex: number, taskSlug: string): void;
+}>();
+
+const { t } = useI18n();
+const databaseStore = useDatabaseStore();
+
+watchEffect(function prepare() {
+  if (props.create) {
+    databaseStore.fetchDatabaseListByProjectId(props.project.id);
+  }
+});
+
+const databaseOfTask = (task: Task | TaskCreate): Database => {
+  if (props.create) {
+    return databaseStore.getDatabaseById((task as TaskCreate).databaseId!);
+  } else {
+    return (task as Task).database!;
+  }
+};
+
+const isSelectedStage = (stage: Stage | StageCreate): boolean => {
+  return stage === props.selectedStage;
+};
+
+const isSelectedTask = (task: Task | TaskCreate): boolean => {
+  return task === props.selectedTask;
+};
+
+const isActiveStage = (stage: Stage | StageCreate): boolean => {
+  if (props.create) return false;
+
+  const task = activeTaskInStage(stage as Stage);
+  if (activeTask(props.pipeline as Pipeline).id === task.id) {
+    return true;
+  }
+  return false;
+};
+
+const isActiveTask = (task: Task | TaskCreate): boolean => {
+  if (props.create) return false;
+  task = task as Task;
+  return activeTask(props.pipeline as Pipeline).id === task.id;
+};
+
+const taskStatusOfStage = (stage: Stage | StageCreate) => {
+  if (props.create) {
```

Comment:
nit: be consistent with other props.create handling without the extra else clause.

```
if (pros.create) {
   return ...
}
xxxx
return ...
```